Fix TestCaddyStartStop

SmilingNavern opened the pull request:
Move this test to caddymain to fix import problems
Minor fixes to bypass locks and race conditions


### 1. What does this change do, exactly?
Fixes TODO test.

### 2. Please link to the relevant issues.
-

### 3. Which documentation changes (if any) need to be made because of this PR?
There shouldn't be documentation changes

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I am willing to help maintain this change if there are issues with it later

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/mholt/caddy?pullRequest=2171) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/mholt/caddy?pullRequest=2171) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/mholt/caddy?pullRequest=2171) it.</sub>

SmilingNavern commented:
Its my first contribution to this project. There is minor change with fixed TODO test. Let me know if this is required change or shouldn't be done in a future.

mholt commented:
Hi @SmilingNavern - welcome to the project, and thanks for submitting a PR!

Can you help me understand, what exactly does this fix? It looks the same to me, except it moved to a different package...

SmilingNavern commented:
Hi, @mholt 

Okay, i will break it down as i understand it:
This test checks if caddy can start. When we use caddyfile and want to start caddy we need to have some plugins(for server_type support). When test was ran at caddy there was no plugins because they should be imported(http plugin for example). And we can't import http into caddy because of cyclic import problem.

So i moved it to caddymain as it seems most logical way to fix this problem.

Also i changed a two lines: we get testInst to correctly close instances(or we get some kind of deadlock) and added sleep to prevent bind problem.

If you need any further explanation let me know.

mholt commented:
Hmm, well this test was disabled because it has kind of a race condition, especially on Windows. It has nothing to do with the plugins. I think I intend to remove integration tests such as this one and put a suite of tests in a separate repository to do them more thoroughly and correctly.

SmilingNavern commented:
Oh, ok. So it wasn't needed after all:) But i've got problems described above after enabling this test.

So should i close this pull request?

mholt commented:
Yeah, sorry. I think a full integration + systems test suite will best be implemented in a new repo.
